core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2CallbackResource.java

```diff
@@ -125,4 +147,9 @@ public Response callback(
         }
         return builder.build();
     }
+
+    private void passErrorToTokenExchange(UUID authId, String format, String... args)
```

Comment:
The two places this is used both have explicit handling of an optional auth id.  I would simplify the caller by doing:
```
private void passErrorToTokenExchange(Optional<UUID> authId, String format, String... args) 
{
    if (tokenExchange.isEmpty() || authId.isEmpty()) {
        return;
    }
    tokenExchange.orElseThrow().setTokenExchangeError(authId.orElseThrow(), format(format, args));
}
```